build/build.bash

```diff
@@ -55,27 +53,36 @@ set timeout 2400
 expect -exact "term% "
 send "ramfs -u -m /n/harvey\n"
 expect -exact "term% "
+send "ramfs -u -m /tmp\n"
+expect -exact "term% "
 send "cd /n/harvey\n"
 expect -exact "term% "
 send "gunzip < /dev/sdC0/data | tar x\n"
 
+expect -exact "term% "
+send "srv -c tcp!10.0.2.2!5640 host /n/host\n"
+
 # Go to the build dir
 expect -exact "term% "
 send "cd /n/harvey/build/scripts\n"
-expect -exact "term% "
-send "rc build && rc repl && rc iso\n"
 
-# and shut down
+# build iso
+expect -exact "term% "
+send "rc build && rc repl && rc usb\n"
 expect -exact "term% "
 send "ls -l /tmp/dist/\n"
 expect -exact "term% "
-send "srv -c tcp!10.0.2.2!5640 host /n/host\n"
+send "time fcp /tmp/plan9-usb.img.bz2 /n/host/\n"
```

Comment:
I still don't understand the reason to write it in the ramfs, read it from the ramfs, write it to a bz2, read the bz2, and write to the 9p server? this results in writing the image to ramfs once, then writing the bz2 to ramfs once, which means reading the image you just wrote again over dog-slow 9p, then reading the bz2 AGAIN and writing it to the 9p server. Further, it requires you to give qemu enough ram to hold the image and the compressed image, since it won't delete the image until it's done compressing ig. why not just tell the command to create the output image in /n/host?

Reply:
I tried that first. I gave up and killed it after replica spent 30 minutes grinding over dog-shit 9p.

Reply:
yes but ... ok I'll time it.
real	12m25.519s
user	0m13.417s
sys	0m7.383s

as written.
with centre:
halting...
done halting
real	11m10.315s
user	0m13.668s
sys	0m7.143s

75 seconds less.

diff --git a/build/scripts/usb b/build/scripts/usb
index 56e0a2d5a..4f3af54ae 100644
--- a/build/scripts/usb
+++ b/build/scripts/usb
@@ -12,7 +12,7 @@ if (test -e /srv/fossil -o -e /srv/fossil.open) {
 sd=sdXX
 dev=/dev/$sd
 target=plan9-usb
-disk=/tmp/$target.img
+disk=/n/host/$target.img
 #gb=1024*1024*1024
 #size=$gb^*1
 size=131072
@@ -103,4 +103,4 @@ rm -f /srv/^(fossil fossil.open fscons fscons.open)
 sleep 8
 kill fossil2 | rc
 
-bzip2 -c9 $disk > $disk.bz2 && rm $disk
+# bzip2 -c9 $disk > $disk.bz2 && rm $disk

diff --git a/build/build.bash b/build/build.bash
index 49477ceac..744d58937 100644
--- a/build/build.bash
+++ b/build/build.bash
@@ -29,7 +29,7 @@ $(cd ..; bash ./build/mkdirs)
 $(cd ..; tar --format ustar --exclude './build' --exclude harvey.tgz --exclude .git --exclude '9legacy.iso*' -czf harvey.tgz *)
 
 expect <<EOF
-spawn qemu-system-i386 -accel kvm -nographic -net user -net nic,model=virtio -m 4096 -vga none -cdrom 9legacy.iso -boot d -hda ../harv.t
+spawn qemu-system-i386 -accel kvm -nographic -net user -net nic,model=virtio -m 8192 -vga none -cdrom 9legacy.iso -boot d -hda ../harv.t
 expect -exact "Selection:"
 send "2\n"
 expect -exact "Plan 9"
@@ -71,12 +71,12 @@ expect -exact "term% "
 send "rc build && rc repl && rc usb\n"
 expect -exact "term% "
 send "ls -l /tmp/dist/\n"
-expect -exact "term% "
-send "time fcp /tmp/plan9-usb.img.bz2 /n/host/\n"
+# expect -exact "term% "
+# send "time fcp /tmp/plan9-usb.img.bz2 /n/host/\n"
 
 # build usb
-expect -exact "term% "
-send "rm -r /tmp/dist/plan9-usb.*\n"
+#expect -exact "term% "
+#send "rm -r /tmp/dist/plan9-usb.*\n"
 expect -exact "term% "
 send "rc iso\n"
 expect -exact "term% "

are you using ufs and is that the suckage?